moqt-client-sample/src/lib.rs

```diff
@@ -521,6 +558,75 @@ impl MOQTClient {
         }
     }
 
+    #[wasm_bindgen(js_name = sendStreamHeaderTrackMessage)]
+    pub async fn send_stream_header_track_message(
+        &self,
+        subscribe_id: u64,
+        track_alias: u64,
+        publisher_priority: u8,
+    ) -> Result<JsValue, JsValue> {
+        let object_stream_writers = self.object_stream_writers.borrow();
+        if let Some(writer) = object_stream_writers.get(&subscribe_id) {
+            let stream_header_track_message =
+                StreamHeaderTrack::new(subscribe_id, track_alias, publisher_priority).unwrap();
+            let mut stream_header_track_message_buf = BytesMut::new();
+            let _ = stream_header_track_message.packetize(&mut stream_header_track_message_buf);
+
+            let mut buf = Vec::new();
+            // Message Type
+            buf.extend(write_variable_integer(
+                u8::from(DataStreamType::StreamHeaderTrack) as u64,
+            ));
+            buf.extend(stream_header_track_message_buf);
+
+            let buffer = js_sys::Uint8Array::new_with_length(buf.len() as u32);
+            buffer.copy_from(&buf);
+            JsFuture::from(writer.write_with_chunk(&buffer)).await
+        } else {
+            return Err(JsValue::from_str("object_stream_writer is None"));
+        }
+    }
+
+    #[wasm_bindgen(js_name = sendObjectStreamTrack)]
+    pub async fn send_object_stream_track(
+        &self,
+        subscribe_id: u64,
+        group_id: u64,
+        object_id: u64,
+        object_payload: Vec<u8>,
+    ) -> Result<JsValue, JsValue> {
+        let object_stream_writers = self.object_stream_writers.borrow();
+        if let Some(writer) = object_stream_writers.get(&subscribe_id) {
+            let object_stream_track =
+                ObjectStreamTrack::new(group_id, object_id, None, object_payload).unwrap();
+            let mut object_stream_track_buf = BytesMut::new();
+            let _ = object_stream_track.packetize(&mut object_stream_track_buf);
+
+            let mut buf = Vec::new();
+            // Message Payload and Payload Length
+            buf.extend(object_stream_track_buf);
+
+            let buffer = js_sys::Uint8Array::new_with_length(buf.len() as u32);
+            buffer.copy_from(&buf);
```

Comment:
`buffer` is used in L618.
So, please move to before L618.
Defining variables just before using is more clear to understand.